src/components/CallView/shared/ReactionToaster.vue

```diff
@@ -0,0 +1,228 @@
+<!--
+  - @copyright Copyright (c) 2023 Maksim Sukharev <antreesy.web@gmail.com>
+  -
+  - @author Maksim Sukharev <antreesy.web@gmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<transition-group name="toast" class="toaster" tag="ul">
+		<li v-for="toast in toasts"
+			:key="toast.seed"
+			class="toast"
+			:style="styled(toast.name, toast.seed)">
+			<span class="toast__reaction">
+				{{ toast.reaction }}
+			</span>
+			<span class="toast__name">
+				{{ toast.name }}
+			</span>
+		</li>
+	</transition-group>
+</template>
+
+<script>
+import Hex from 'crypto-js/enc-hex.js'
+import SHA1 from 'crypto-js/sha1.js'
+
+import { subscribe, unsubscribe } from '@nextcloud/event-bus'
+
+import usernameToColor from '@nextcloud/vue/dist/Functions/usernameToColor.js'
+
+export default {
+	name: 'ReactionToaster',
+
+	props: {
+		/**
+		 * The conversation token
+		 */
+		token: {
+			type: String,
+			required: true,
+		},
+		callParticipantModels: {
+			type: Array,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			registeredModels: {},
+			reactionsQueue: [],
+			intervalId: null,
+			animationLength: 2000,
+			toasts: [],
+		}
+	},
+
+	computed: {
+		participants() {
+			return this.$store.getters.participantsList(this.token)
+		},
+	},
+
+	watch: {
+		callParticipantModels(models) {
+			// subscribe connected models for reaction signals
+			const addedModels = models.filter(model => !this.registeredModels[model.attributes.peerId])
+			addedModels.forEach(addedModel => {
+				this.registeredModels[addedModel.attributes.peerId] = addedModel
+				this.registeredModels[addedModel.attributes.peerId].on('reaction', this.handleReaction)
+			})
+
+			// unsubscribe disconnected models
+			const removedModels = Object.keys(this.registeredModels).filter(registeredModelId => !models.find(model => model.attributes.peerId === registeredModelId))
+			removedModels.forEach(removedModel => {
+				this.registeredModels[removedModel].off('reaction', this.handleReaction)
+				delete this.registeredModels[removedModel]
+			})
+		},
+	},
+
+	mounted() {
+		this.intervalId = setInterval(this.processReactionsQueue, this.animationLength / 4)
+		subscribe('send-reaction', this.handleOwnReaction)
+	},
+
+	beforeDestroy() {
```

Comment:
The registered models should be iterated over and deregistered (`.off('reaction', this.handleReaction)`) when the component is destroyed (or before being destroyed, that should not matter).

The models will be (probably) deregistered anyway when the models are removed from `callParticipantModels` even if the component was already destroyed (although I am not fully sure about that, I do not know if watchers are still run after the component is destroyed), but it would be cleaner to do it when the component is destroyed and thus no longer needed.